File: src/Go/functions_test.go
package Go

import (
	"testing"
)

func TestPalindrome(t *testing.T) {
	if !IsPalindrome("racecar") {
		t.Fatalf("racecar is a palindrome")
	}
}

func TestNth(t *testing.T) {
	arr := []int{1, 2, 3, 4, 5}
	if Nthmax(arr, 1) != 4 {
		t.Fatalf("Fail!")
	}
}

func TestFreq(t *testing.T) {
	if Freq("racecarrr") != "r" {
		t.Fatalf("Fail!")
	}
}

func TestFindMax(t *testing.T) {
	arr := []int{1, 2, 3, 4, 5}
	if FindMax(arr) != 5 {
		t.Fatalf("Fail!")
	}
}

func TestCountOnes(t *testing.T) {
	arr := []int{1, 2, 3, 4, 5}
	if CountOnes(arr) != 1 {
		t.Fatalf("Fail!")
	}
}

func TestRevTup(t *testing.T) {
	arr := [3]int{1, 2, 3}
	answ := [3]int{3, 2, 1}
	if RevTup(arr) != answ {
		t.Fatalf("Fail!")
	}
}

func TestIsEven(t *testing.T) {
	if IsEven(1) {
		t.Fatalf("Fail!")
	}
}

func TestVolume(t *testing.T) {
	arr1 := [3]int{6, 3, 7}
	arr2 := [3]int{4, 1, 3}
	if Volume(arr1, arr2) != 16 {
		t.Fatalf("Fail!")
	}
}

func TestFibonacci(t *testing.T) {
	if Fibonacci(10) != 34 {
		t.Fatalf("Fail!")
	}
}

func TestCountOccurence(t *testing.T) {
	arr := []int{1, 2, 3, 4, 5}
	if CountOcc(arr, 1) != 1 {
		t.Fatalf("Fail!")
	}
}

func TestToBinstring(t *testing.T) {
	if ToBinstring(19) != "10011" {
		t.Fatalf("Fail!")
	}
}

func TestSubset(t *testing.T) {
	arr1 := []int{6, 3, 7}
	arr2 := []int{4, 1, 3}
	if subset(arr1, arr2) {
		t.Fatalf("Fail!")
	}
}

File: src/JavaScript/functions.test.js
const { isPalindrome, nthmax, subset, to_binstring, count_occ, fibonacci, volume, is_even, rev_tup, count_ones, find_max, freq } = require("./functions")

test('palindrome works properly', () => {
    expect(isPalindrome("racecar")).toBe(true)
})

test('nthmax', () => {
    expect(nthmax(1, [1,2,3,4,5])).toBe(4)
})

test('subset', () => {
    expect(subset([1,2,3], [4,5,6])).toBe(false)
})

test('to_binstring', () => {
    expect(to_binstring(19)).toBe("10011")
})

test('count_occ', () => {
    expect(count_occ([1,2,3,4,1,2,2,1,1], 1)).toBe(4)
})

test('fibonacci', () => {
    expect(fibonacci(10)).toBe(34)
})

test('volume', () => {
    expect(volume([6,3,7], [4,1,3])).toBe(16)
})

test('is_even', () => {
    expect(is_even(3)).toBe(false)
})

test('rev_tup', () => {
    expect(rev_tup([1,2,3])).toStrictEqual([3,2,1])
})

test('count_ones', () => {
    expect(count_ones([1,2,3,4,1,2,2,1,1])).toBe(4);
})

test('find_max', () => {
    expect(find_max([1,2,3,4,1,2,2,1,1])).toBe(4)
})

test('freq', () => {
    expect(freq("scabbdasbdasdeghoubb")).toBe("b")
})